sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/event_processor.py

```diff
@@ -4,190 +4,85 @@
-         and asynchronously start receiving EventData from EventHub and processing events.
-
-        :return: None
-
-        """
-        log.info("EventProcessor %r is being started", self._id)
-        ownership_manager = OwnershipManager(self._eventhub_client, self._consumer_group_name, self._id,
-                                             self._partition_manager, self._ownership_timeout)
-        if not self._running:
-            self._running = True
-            while self._running:
-                try:
-                    claimed_ownership_list = await ownership_manager.claim_ownership()
-                except Exception as err:  # pylint:disable=broad-except
-                    log.warning("An exception (%r) occurred during balancing and claiming ownership for eventhub %r "
-                                "consumer group %r. Retrying after %r seconds",
-                                err, self._eventhub_name, self._consumer_group_name, self._polling_interval)
-                    await asyncio.sleep(self._polling_interval)
-                    continue
-
-                if claimed_ownership_list:
-                    claimed_partition_ids = [x["partition_id"] for x in claimed_ownership_list]
-                    to_cancel_list = self._tasks.keys() - claimed_partition_ids
-                    self._create_tasks_for_claimed_ownership(claimed_ownership_list)
-                else:
-                    to_cancel_list = set(self._tasks.keys())
-                    log.info("EventProcessor %r hasn't claimed an ownership. It keeps claiming.", self._id)
-                if to_cancel_list:
-                    self._cancel_tasks_for_partitions(to_cancel_list)
-                    log.info("EventProcesor %r has cancelled partitions %r", self._id, to_cancel_list)
-                await asyncio.sleep(self._polling_interval)
-
-    async def stop(self):
-        """Stop the EventProcessor.
-
-        The EventProcessor will stop receiving events from EventHubs and release the ownership of the partitions
-        it is working on.
-        Other running EventProcessor will take over these released partitions.
-
-        A stopped EventProcessor can be restarted by calling method `start` again.
-
-        :return: None
-
-        """
-        self._running = False
-        for _ in range(len(self._tasks)):
-            _, task = self._tasks.popitem()
-            task.cancel()
-        log.info("EventProcessor %r has been cancelled", self._id)
-        await asyncio.sleep(2)  # give some time to finish after cancelled.
+    def _get_last_enqueued_event_properties(self, partition_id):
+        if partition_id in self._tasks and partition_id in self._last_enqueued_event_properties:
+            return self._last_enqueued_event_properties[partition_id]
+        raise ValueError("You're not receiving events from partition {}".format(partition_id))
```

Comment:
What's the intent here? That `partition_id` was incorrect, or something else? Doesn't seem clear to me

Reply:
Not blocking for preview5, solving